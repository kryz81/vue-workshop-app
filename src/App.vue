<template>
  <div class="h-100 main" :class="mode">
    <Header />
    <b-container>
      <b-row>
        <b-col>
          <router-view />
        </b-col>
      </b-row>
    </b-container>
  </div>
</template>

<script>
import { mapState } from "vuex";
import Header from "./components/Header";

export default {
  components: {
    Header
  },
  computed: {
    ...mapState(["mode"])
  }
};
</script>

<style scoped>
.main.dark {
  background-color: #777;
  color: #fff;
}

.main.light {
  background-color: #fff;
  color: #000;
}
</style>
