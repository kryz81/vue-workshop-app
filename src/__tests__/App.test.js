import { shallowMount } from "@vue/test-utils";
import App from "../App";

describe("App", () => {
  it("renders correctly", () => {
    const wrapper = shallowMount(App, {
      stubs: ["router-view", "b-container", "b-row", "b-col"],
      computed: { mode: () => "dark" }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
