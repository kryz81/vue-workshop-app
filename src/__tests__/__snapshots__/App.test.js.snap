// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div class="h-100 main dark">
  <header-stub></header-stub>
  <b-container-stub>
    <b-row-stub>
      <b-col-stub>
        <router-view-stub></router-view-stub>
      </b-col-stub>
    </b-row-stub>
  </b-container-stub>
  <b-container-stub>
    <b-row-stub>
      <b-col-stub>
        <router-view-stub name="footer"></router-view-stub>
      </b-col-stub>
    </b-row-stub>
  </b-container-stub>
</div>
`;
